examples/pub_sub/src/main.rs

```diff
@@ -1,28 +1,24 @@
-mod event_bus;
+mod msg_ctx;
 mod producer;
 mod subscriber;
 
 use producer::Producer;
 use subscriber::Subscriber;
-use yew::{html, Component, Context, Html};
+use yew::prelude::*;
 
-pub struct Model;
+use msg_ctx::{Message, MessageContext};
 
-impl Component for Model {
-    type Message = ();
-    type Properties = ();
+#[function_component(Model)]
```

Comment:
I wonder if we would want to encourage making Provider components that just to prodive the context to children like
```jsx
<MessageProvider>
            <Producer />
            <Subscriber />
</MessageProvider>
```
and thus make examples do that as well